### PR TITLE
Cleanup of demons

### PR DESCRIPTION
**Pull request by lolgear**
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none
* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ...

### Pull Request Description

Demos are now free from CocoaPods reign.

### CODE REVIEW

File: Benchmarking/DynamicLogging.m [hunk 1]
```diff
@@ -7,7 +7,8 @@
 
 #import "DynamicLogging.h"
 #import "PerformanceTesting.h"
-#import "DDLogMacros.h"
+//#import "DDLogMacros.h"
```
Comment (ffried) [inline Benchmarking/DynamicLogging.m]: Just remove the line if it's no longer needed.

File: Benchmarking/StaticLogging.m [hunk 2]
```diff
@@ -7,7 +7,8 @@
 
 #import "StaticLogging.h"
 #import "PerformanceTesting.h"
-#import "DDLogMacros.h"
+//#import "DDLogMacros.h"
```
Comment (ffried) [inline Benchmarking/StaticLogging.m]: Same here.

Comment (lolgear) [reply Benchmarking/StaticLogging.m]: @ffried 

Proof of the idea first :)

Comment (lolgear) [reply Benchmarking/StaticLogging.m]: Next - clean dirty places :)

**Comment by ffried** [comment]
> Demos are now free from CocoaPods reign.

I see, I'm not the only one who doesn't like CocoaPods 😜 
But I think maybe we should keep one Demo that actually uses CocoaPods as reference. Also, so that we see if there's any breakage when integrating with CocoaPods...

**Comment by lolgear** [comment]
@ffried 

Well, they are "maybe" necessary for One-day apps. 
But, well, hm, for 5+ years (?) large frameworks with 10k stars... It is a bit luxury.

Well, hm, you have `pod lib lint` for it.

Also, I suggest not to put `Pods` and `Carthage` into one place with `Demos`.
Demos are for people who want "run and play".
I always hate "Pods.xcodeproj" in red color, when I only want to check "what is going on in this example".

**Comment by ffried** [comment]
> Well, hm, you have `pod lib lint` for it.

and

> Also, I suggest not to put Pods and Carthage into one place with Demos.

are both fair points!
My idea would anyways more go into the direction of `Integration` or even `Tests`.

**Comment by lolgear** [comment]
@ffried 

Well, I would like to cleanup `Tests` too..

File: Benchmarking/DynamicLogging.m (hunk 1 above)
Comment (ffried) [reply Benchmarking/DynamicLogging.m]: @lolgear Did you miss this one in your cleanup commit?

**Comment by lolgear** [comment]
@ffried 

I suggest to add `Deployment` folder where CocoaPods and Carthage fight against each other.

**Comment by ffried** [comment]
Looks good to  me now. LGTM once Travis is done.

**Comment by lolgear** [comment]
@ffried 

How I see these directories:

`Demos` - examples for people with only mouse - hit on button run and see results.
`Tests` - borders and corners of behavior.
`Integration` - correctness of behavior in ecosystem. ( Apple or not? )
`Deployment` - correctness of deployment through other channels (CocoaPods/Carthage/Submodules). ( Maybe even separate repository? )

**Comment by ffried** [comment]
@lolgear 
I mostly agree. Although Demos should also contain some advanced stuff. Like "I want to do this super advanced thing, what do the library owners recommend to do that?".

**Comment by ffried** [comment]
(which is, what some of the demos do today already)

**Comment by lolgear** [comment]
@ffried 

`They way you deliver pizza is not a way you cook it.`

**Comment by LumberjackPolice** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by lolgear** [comment]
@ffried 

So...
Could it be merged?

**Comment by ffried** [comment]
@lolgear Wayyyy ahead of you... 😜

**Comment by lolgear** [comment]
@ffried 
Too fast :D

**Pull request merged**